pykickstart/commands/timesource.py

```diff
@@ -0,0 +1,149 @@
+#
+from pykickstart.errors import KickstartParseError
+from pykickstart.version import F33
+from pykickstart.base import BaseData, KickstartCommand
+from pykickstart.options import KSOptionParser
+
+from pykickstart.i18n import _
+
+class F33_TimesourceData(BaseData):
+    removedKeywords = BaseData.removedKeywords
+    removedAttrs = BaseData.removedAttrs
+
+    def __init__(self, *args, **kwargs):
+        BaseData.__init__(self, *args, **kwargs)
+        self.ntp_server = kwargs.get("ntp_server", "")
+        self.ntp_pool = kwargs.get("ntp_pool", "")
+        self.ntp_disable = kwargs.get("ntp_disable", False)
+        self.nts = kwargs.get("nts", False)
+
+    def __eq__(self, y):
+        if not y:
+            return False
+
+        if self.ntp_server:
+            return self.ntp_server == y.ntp_server and self.nts == y.nts
+        elif self.ntp_pool:
+            return self.ntp_pool == y.ntp_pool and self.nts == y.nts
+        else:
+            # given that one of ntp_server/ntp_pool/ntp_disable is required
+            # this will be --ntp_disable & such timesource invocations will
+            # always be the same by definition
+            return True
+
+    def __ne__(self, y):
+        return not self == y
+
+    def __str__(self):
+        retval = BaseData.__str__(self)
+
+        args = self._getArgsAsStr()
+        if args:
+            retval += "timesource%s\n" % args
+
+        return retval
+
+    def _getArgsAsStr(self):
+        retval = ""
+
+        if self.ntp_server:
+            retval += " --ntp-server=%s" % self.ntp_server
+        if self.ntp_pool:
+            retval += " --ntp-pool=%s" % self.ntp_pool
+        if self.nts:
+            retval += " --nts"
+        if self.ntp_disable:
+            retval += " --ntp-disable"
+
+        return retval
+
+
+class F33_Timesource(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+        self.timesource_list = kwargs.get("timesource_list", [])
+        self.exclusive_required_options = [("ntp-server", "--ntp-server"),
+                                           ("ntp-pool", "--ntp-pool"),
+                                           ("ntp-disable", "--ntp-disable")]
+
+    def __str__(self):
+        retval = ""
+        for timesource_command in self.timesource_list:
+            retval += timesource_command.__str__()
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="timesource", description="""
+                            Configures a timesource.""",
+                            version=F33)
+        op.add_argument("--ntp-server", version=F33, help="""
+                        A single NTP server.
+
+                        ``--ntp-server`` and ``--ntp-pool`` are mutually exclusive.
```

Comment:
I guess that they should be also mutually exclusive with `--ntp-disable`.

Reply:
Fixed.